vpd-manager/src/worker.cpp

```diff
@@ -1430,6 +1449,24 @@ std::tuple<bool, std::string>
         m_activeCollectionThreadCount++;
         m_mutex.unlock();
 
+        // Set CollectionStatus as InProgress. Since it's an intermediate state
+        // D-bus set-property call is good enough to update the status.
+        try
+        {
+            const std::string& l_collStatusProp = "CollectionStatus";
```

Comment:
Why a variable is needed here? Its a one time use variable. Can be used in place?

Reply:
dbusUtility::writeDbusProperty() expects ref of the property variable. so I need a memory address to be allocated

Reply:
That's ok, it anyways accepts constant reference.

Reply:
done